backend/src/services/coachService.ts

```diff
@@ -0,0 +1,86 @@
+import CustomError from "../middlewares/customError";
+import User from "../models/User";
+import { NOT_ALLOWED, USER_NOT_FOUND } from "../utils/errorCodes";
+
+export const getCoachesService = async (
+  role: "ADMIN" | "COACH" | "TRAINEE",
+  {
+    searchString,
+    sortBy,
+    coachesPerPage,
+  }: { searchString: string; sortBy: string; coachesPerPage: number },
+) => {
+  if (role !== "ADMIN") {
+    throw new CustomError(NOT_ALLOWED, "Only admins can view coaches", 403);
+  }
+
+  const coaches = await User.aggregate([
```

Comment:
For the time being we can keep this here but when we'll see the need we'll create the repository layer were all the complex data access methods will reside. So please keep an eye on this and when we see that the services are polluted with too many complex queries then we should start moving things to the repository/data access layer :)